### DISCUSSION TITLE
My experience with Copilot

### DISCUSSION THREAD

**Discussion by Shybella** [community/community#119717] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

Context embeds are jank with Copilot. It doesn't understand the whole context when I know it can. OpenAI API GPT4 requests with embeddings understand the context just fine. You guys are doing something wrong and I don't know what. 

Response times are insane, sometimes up to 10 minutes. ChatGPT and OpenAI API/playground do NOT have this issue. This is an edge case related to this product only. 

IDE support, especially for intellij is subpar (few free plugins do better). I cannot speak for other IDEs. 

The most frustrating feature is the response times, it makes the product completely useless. I outpace any assistant qualities. 

As of this year, this product is my worst purchase. I also purchased back in 2022/23 but the response time wasn't much of an issue and it was such a new product that IDE support was in its infantancy. Context/embeds also did not exist back then. 

tl;dr Github is falsely advertising a product and also compounding it with a bad experience.  

Follow-up opinion:
All these code assistants are gimmicks or flat out do not work and just pose as a scam rather than something innovative and useful. 
Maybe someone can change my mind on the "all" part but my bias outweighs what I have experienced.
